vllm/v1/core/kv_cache_manager.py

```diff
@@ -198,11 +216,34 @@ def get_computed_blocks(self, request: Request) -> tuple[KVCacheBlocks, int]:
         # num_computed_tokens to be block-size aligned. Removing this limitation
         # could slightly improve performance in the future.
         max_cache_hit_length = request.num_tokens - 1
-        computed_blocks, num_new_computed_tokens = (
-            self.coordinator.find_longest_cache_hit(
-                request.block_hashes, max_cache_hit_length
+
+        # Optimization: Reuse previous result when consecutive requests are the same.
+        if (
+            request.block_hashes
+            and self._cache_hit_cache is not None
+            and max_cache_hit_length == self._cache_hit_cache.max_cache_hit_length
+            and len(request.block_hashes) == self._cache_hit_cache.num_block_hashes
+            and request.block_hashes[-1] == self._cache_hit_cache.last_block_hash
+        ):
```

Comment:
We can reuse the computed_blocks of the previous request only when we can ensure these blocks are not evicted.
1. If the model only has full attention, I guess it's fine but want some proof on it
2. If the model has sliding window attention, I feel there may be some problem as the previous request may free the early blocks outside the sliding window and make these blocks evocable.